### PR TITLE
Add Discord auth provider

### PR DESCRIPTION
**Pull request by derrickmehaffy**
My PR is a:
🚀 New feature

Main update on the:
Plugin

Adding Discord oAuth2 as a provider

**note:** due to the username not being unique on discord I joined the username + 4 digit discriminator to make it so, more details can be found here: [Discord Docs](https://discordapp.com/developers/docs/resources/user#user-object)

If this isn't the correct way to handle it let me know. :smiley:

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/services/Providers.js [hunk 1]
```diff
@@ -109,6 +109,40 @@ const getProfile = async (provider, query, callback) => {
...
+            }
+          }
+        }
+      });
+      discord.query().get('users/@me').auth(access_token).request((err, res, body) => {
+        if (err) {
+          callback(err);
+        } else {
+          // Combine username and discriminator because discord username is not unique
+          var username = body.username + '#' + body.discriminator;
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-users-permissions/services/Providers.js]: A typical Discord "username" looks something like `DMehaffy#1337` hence the addition of the `#`

**Comment by synth3tk** [comment]
I've tested this implementation and can confirm that it works! :+1:

**Comment by derrickmehaffy** [comment]
Side note of things that might need to be done (If @Aurelsicoko approves):

- [ ] Create react test (like the other providers)
- [ ] Update Strapi Documentation

**Comment by Aurelsicoko** [comment]
@derrickmehaffy The code looks good to me but as you noticed, can you please update the documentation?

**Comment by derrickmehaffy** [comment]
@Aurelsicoko is there anything specific you want to see in the documentation?

**Comment by derrickmehaffy** [comment]
Created PR for example react app, note that it doesn't have the font awesome discord icon

https://github.com/strapi/strapi-examples/pull/36

**Pull request merged**